### PR TITLE
Add relationship predicates (anyIn and anyEquals) and stringEquals

### PR DESCRIPTION
**Pull request by djyates**
Additional predicate initialisers and unit tests, including:

`NSPredicate(format: "ANY %K == %@", argumentArray: [keyPath, value])`
`NSPredicate(format: "ANY %K IN %@", argumentArray: [keyPath, value])`
`NSPredicate(format: "SELF ==[cd] %@", argumentArray: [string])`

Also added a flag to allow case sensitive comparison of strings

### CODE REVIEW

File: PeakCoreDataTests/PredicateTests.swift [hunk 1]
```diff
@@ -186,67 +186,190 @@ class PredicateTests: XCTestCase {
...
         let keyPathPredicate = NSPredicate(stringBeginsWith: testString, keyPath: #keyPath(Object.name))
-        XCTAssertFalse(keyPathPredicate.evaluate(with: jam))
-        XCTAssertTrue(keyPathPredicate.evaluate(with: elephants))
-        XCTAssertFalse(keyPathPredicate.evaluate(with: peanuts))
+        XCTAssertFalse(keyPathPredicate.evaluate(with: object1))
+        XCTAssertTrue(keyPathPredicate.evaluate(with: object2))
+        XCTAssertFalse(keyPathPredicate.evaluate(with: object3))
+    }
+    
+    func testStringBeginsWithCaseSsensitive() {
```
Comment (BendyWalker) [inline PeakCoreDataTests/PredicateTests.swift]: Typo 🐍
```suggestion
    func testStringBeginsWithCaseSensitive() {
```

File: PeakCoreDataTests/PredicateTests.swift [hunk 2]
```diff
@@ -186,67 +186,190 @@ class PredicateTests: XCTestCase {
...
+        XCTAssertTrue(selfPredicate.evaluate(with: object2.name))
+        XCTAssertFalse(selfPredicate.evaluate(with: object3.name))
+        
+        let keyPathPredicate = NSPredicate(stringEquals: testString, keyPath: #keyPath(Object.name))
+        XCTAssertFalse(keyPathPredicate.evaluate(with: object1))
+        XCTAssertTrue(keyPathPredicate.evaluate(with: object2))
+        XCTAssertFalse(keyPathPredicate.evaluate(with: object3))
+    }
+    
+    func testStringEqualsCaseSsensitive() {
```
Comment (BendyWalker) [inline PeakCoreDataTests/PredicateTests.swift]: Typo 🐍
```suggestion
    func testStringEqualsCaseSensitive() {
```

File: PeakCoreDataTests/PredicateTests.swift [hunk 3]
```diff
@@ -186,67 +186,190 @@ class PredicateTests: XCTestCase {
...
         let keyPathPredicate = NSPredicate(stringEndsWith: testString, keyPath: #keyPath(Object.name))
-        XCTAssertFalse(keyPathPredicate.evaluate(with: jam))
-        XCTAssertTrue(keyPathPredicate.evaluate(with: elephants))
-        XCTAssertFalse(keyPathPredicate.evaluate(with: peanuts))
+        XCTAssertFalse(keyPathPredicate.evaluate(with: object1))
+        XCTAssertTrue(keyPathPredicate.evaluate(with: object2))
+        XCTAssertFalse(keyPathPredicate.evaluate(with: object3))
+    }
+    
+    func testStringEndsWithCaseSsensitive() {
```
Comment (BendyWalker) [inline PeakCoreDataTests/PredicateTests.swift]: Typo 🐍
```suggestion
    func testStringEndsWithCaseSensitive() {
```

**Review by BendyWalker — CHANGES_REQUESTED** [review]
Apart from the typos, these changes look good. Presumably the case insensitive stuff could be used to indirectly help with #12?

**Review by BendyWalker — APPROVED** [review]

**Review by SamTTaylor — APPROVED** [review]

**Review by blork — APPROVED** [review]

**Pull request merged**